src/lib/firestore/post.ts

```diff
@@ -0,0 +1,62 @@
+import { DocumentData, QueryDocumentSnapshot } from 'firebase-admin/firestore'
+import moment from 'moment'
+
+import { db } from './firestore'
+
+import { postStatus } from '@/consts/constants'
+import { Post } from '@/types/Post'
+
+export const GetPosts = async () => {
+  let post: Post[] = []
+  ;(
+    await db
+      .collection('posts')
+      .where('status', '==', postStatus.APPROVED)
+      .withConverter<Post>(postConverter)
```

Comment:
承認日時の降順でソートしたいです。
```suggestion
      .collection('posts')
      .where('status', '==', postStatus.APPROVED)
      .orderBy('approved', 'desc')
      .withConverter<Post>(postConverter)
```

項目を足しました。こちら承認した時の更新時にタイムスタンプを記録してください。
https://github.com/foodbank-ymg/thanks-web-backend/wiki/Schema/_compare/ebafb8c7e3283e9749a2336f091f914e261f20f5...fdafb55698ab11f51d4a832c7e95604d7bb9136d

おそらくローカルで実行すると、複合インデクス（status & approvedAt）がない旨のエラーが出ると思います。
そのエラーに、その複合インデクスを作成するリンクが付いてると思います。
そのままブラウザで開いて、指示通りのインデクスを作成してください。

Reply:
orderBy('approvedAt', 'desc')でしょうか。